### PR TITLE
[zh-cn] Cleanup custom-resource-definition-versioning page

### PR DESCRIPTION
**Pull request by Zhuzhenghao**
This pr removes redundant English documentation and fixs a link in custom-resource-definition-versioning page.

### CODE REVIEW

**Comment by tengqm** [comment]
This page has some new changes in English page that should be sync'ed, although this doesn't necessarily mean the translated text needs a change.
/hold

**Comment by Zhuzhenghao** [comment]
I have resync'ed the file just now.

But I just found one different section, [see here](https://github.com/kubernetes/website/pull/39079/files#diff-314268937dfef5e00390f3d1a8cc27b852064e865e0cb633176bc84ea8e89b36R638-R641).

And I also resync'ed some spaces at the end of statements. Btw, is it necessary to remove these spaces?

**Comment by Zhuzhenghao** [comment]
我发现注释的英文部门的最后一行偶尔会出现一个空格，我想了想，会不是这样的操作引起的：

我们用了编辑器的注释快捷键，`cmd+/`之后，从`-->`的最前面换行，导致遗留下 `to` 后面的空格？



当然，这似乎不是很重要 😅

Then, we

**Comment by tengqm** [comment]
> 

We basically avoid any trailing spaces in the markdown files.

File: content/zh-cn/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md [hunk 1]
```diff
@@ -569,7 +565,7 @@ API 服务器还支持在需要转换时调用外部服务的 webhook 转换。
 * 定制资源的 PUT 请求所针对版本与存储版本不同。
 
 <!--
-To cover all of these cases and to optimize conversion by the API server,
+To cover all of these cases and to optimize conversion by the API server,
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:568]: These spaces should be trimmed.

File: content/zh-cn/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md [hunk 2]
```diff
@@ -627,7 +623,7 @@ how to [authenticate API servers](/docs/reference/access-authn-authz/extensible-
 A conversion webhook must not mutate anything inside of `metadata` of the converted object
 other than `labels` and `annotations`.
 Attempted changes to `name`, `UID` and `namespace` are rejected and fail the request
-which caused the conversion. All other changes are ignored. 
+which caused the conversion. All other changes are ignored.
```
Comment (tengqm) [inline content/zh-cn/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md:626]: Same here.

**Review by tengqm — COMMENTED** [review]
When running the following command, I can see more changes than what you have sync'ed:

```shell
./scripts/lsync.sh content/zh-cn/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning.md
```

**Comment by Zhuzhenghao** [comment]
Changes that I ran the command are as follows:

### Part  1

```
@@ -78,8 +78,8 @@ Removing an old version:
    If this occurs, switch back to using `served:true` on the old version, migrate the 
    remaining clients to the new version and repeat this step.
 1. Ensure the [upgrade of existing objects to the new stored version](#upgrade-existing-objects-to-a-new-stored-version) step has been completed.
-    1. Verify that the `storage` is set to `true` for the new version in the `spec.versions` list in the CustomResourceDefinition.
-    1. Verify that the old version is no longer listed in the CustomResourceDefinition `status.storedVersions`.
+   1. Verify that the `storage` is set to `true` for the new version in the `spec.versions` list in the CustomResourceDefinition.
+   1. Verify that the old version is no longer listed in the CustomResourceDefinition `status.storedVersions`.
 1. Remove the old version from the CustomResourceDefinition `spec.versions` list.
 1. Drop conversion support for the old version in conversion webhooks.
```

Synced in [here](https://github.com/kubernetes/website/pull/38592/files#diff-314268937dfef5e00390f3d1a8cc27b852064e865e0cb633176bc84ea8e89b36R138-R139).


### Part 2
```
@@ -1037,7 +1037,7 @@ What happens to the object that is being returned while serving the read
 request depends on what is specified in the CRD's `spec.conversion`:
 - if the default `strategy` value `None` is specified, the only modifications
   to the object are changing the `apiVersion` string and perhaps [pruning
-  unknown fields](/docs/concepts/extend-kubernetes/api-extension/custom-resources/custom-resource-definitions/#field-pruning)
+  unknown fields](/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#field-pruning)
   (depending on the configuration). Note that this is unlikely to lead to good
   results if the schemas differ between the storage and requested version.
   In particular, you should not use this strategy if the same data is
```

Synced in my [PR](https://github.com/kubernetes/website/pull/39079/files#diff-314268937dfef5e00390f3d1a8cc27b852064e865e0cb633176bc84ea8e89b36R1384).


### Part 3
```
@@ -1051,19 +1051,19 @@ one version to another.
 
 To illustrate this, consider the following hypothetical series of events:
 
-1.  The storage version is `v1beta1`. You create an object. It is stored at version `v1beta1`
-2.  You add version `v1` to your CustomResourceDefinition and designate it as
-    the storage version. Here the schemas for `v1` and `v1beta1` are identical,
-    which is typically the case when promoting an API to stable in the
-    Kubernetes ecosystem.
-3.  You read your object at version `v1beta1`, then you read the object again at
-    version `v1`. Both returned objects are identical except for the apiVersion
-    field.
-4.  You create a new object. It is stored at version `v1`. You now
-    have two objects, one of which is at `v1beta1`, and the other of which is at
-    `v1`.
-5.  You update the first object. It is now stored at version `v1` since that
-    is the current storage version.
+1. The storage version is `v1beta1`. You create an object. It is stored at version `v1beta1`
+2. You add version `v1` to your CustomResourceDefinition and designate it as
+   the storage version. Here the schemas for `v1` and `v1beta1` are identical,
+   which is typically the case when promoting an API to stable in the
+   Kubernetes ecosystem.
+3. You read your object at version `v1beta1`, then you read the object again at
+   version `v1`. Both returned objects are identical except for the apiVersion
+   field.
+4. You create a new object. It is stored at version `v1`. You now
+   have two objects, one of which is at `v1beta1`, and the other of which is at
+   `v1`.
+5. You update the first object. It is now stored at version `v1` since that
+   is the current storage version.
 
 ### Previous storage versions
```

Synced in [here](https://github.com/kubernetes/website/pull/38592/files#diff-314268937dfef5e00390f3d1a8cc27b852064e865e0cb633176bc84ea8e89b36R1414-R1435).


### Part4
```
@@ -1079,19 +1079,19 @@ procedure.
 
 *Option 1:* Use the Storage Version Migrator
 
-1.  Run the [storage Version migrator](https://github.com/kubernetes-sigs/kube-storage-version-migrator)
-2.  Remove the old version from the CustomResourceDefinition `status.storedVersions` field.
+1. Run the [storage Version migrator](https://github.com/kubernetes-sigs/kube-storage-version-migrator)
+2. Remove the old version from the CustomResourceDefinition `status.storedVersions` field.
 
 *Option 2:* Manually upgrade the existing objects to a new stored version
 
 The following is an example procedure to upgrade from `v1beta1` to `v1`.
 
-1.  Set `v1` as the storage in the CustomResourceDefinition file and apply it
-    using kubectl. The `storedVersions` is now `v1beta1, v1`.
-2.  Write an upgrade procedure to list all existing objects and write them with
-    the same content. This forces the backend to write objects in the current
-    storage version, which is `v1`.
-3.  Remove `v1beta1` from the CustomResourceDefinition `status.storedVersions` field.
+1. Set `v1` as the storage in the CustomResourceDefinition file and apply it
+   using kubectl. The `storedVersions` is now `v1beta1, v1`.
+2. Write an upgrade procedure to list all existing objects and write them with
+   the same content. This forces the backend to write objects in the current
+   storage version, which is `v1`.
+3. Remove `v1beta1` from the CustomResourceDefinition `status.storedVersions` field.
 
 {{< note >}}
 The flag `--subresource` is used with the kubectl get, patch, edit, and replace commands to
```

Synced in [here](https://github.com/kubernetes/website/pull/38592/files#diff-314268937dfef5e00390f3d1a8cc27b852064e865e0cb633176bc84ea8e89b36R1466-R1496)


Am I not sure I'm missing something?

**Comment by tengqm** [comment]
Thanks for double checking.
/lgtm
/approve

**Pull request merged**